x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx

```diff
@@ -66,6 +67,18 @@ const formatMetric = (type: InfraAssetMetricType, value: number | undefined | nu
   return value || value === 0 ? createInventoryMetricFormatter({ type })(value) : 'N/A';
 };
 
+const buildMetricCell = (
+  value: number,
+  formatType: InfraAssetMetricType,
+  hasSystemMetrics?: boolean
+) => {
+  if (!hasSystemMetrics && value === 0) {
```

Comment:
Could we check if `value` is `null`? 0 can be a valid value. 

eg: An APM host is collecting CPU usage, but for whatever reason the value is 0. In this case, we should not show the badge. But if for another APM host, CPU usage is not being collected, thus the value is null. For this host, we should show the badge

Reply:
I was thinking about commenting on that but I saw that the value is of type `number` and I thought we would consider `0` as invalid and `null` values are converted to `0`s ( but it could be valid also if it's a short time range/ no metrics collected for a period or somehow is actually `0`) 

We should also make sure we send `null` and not `0` in case it's a missing value and change the type to `number | null`

Reply:
While working on this, I had the same thought, as the AC specifically mentions the `null` case and not the `0` value.

However, at that point in the code, the value will never be `null` because we convert it to `0` if it doesn’t exist (as @jennypavlova says).

<img width="646" alt="Screenshot 2024-08-26 at 10 59 14" src="https://github.com/user-attachments/assets/564661eb-27a5-4140-8d6c-2afb9dbbf22f">


I tried not forcing this mapping to `0` to properly handle `null` values, but then the table sorting by columns doesn’t work correctly because it can't handle `null` values in sorting. How do you all think sorting should handle these `N/A` badges?


To clarify, cases where `hasSystemMetrics === false` with a `0` value coming from the API should not display as `N/A` because there might be cases where we want to show `0`, correct? 

For example, an APM host can collect CPU usage and have `hasSystemMetrics === false`.
These screenshots have CPU usage at `60`, but it could happen that we receive a `0` as value there instead.

|Table|API|
|-|-|
|![Screenshot 2024-08-26 at 11 06 56](https://github.com/user-attachments/assets/81d3aeb4-5f64-4d7d-a236-2696f60b3daf)|![Screenshot 2024-08-26 at 11 08 46](https://github.com/user-attachments/assets/cc9cc13c-b8ca-44f8-81f6-d8a8b422591b)|

Reply:
To give a bit more visibility on how the table sorting works when we allow `null` values (by not mapping them to `0`).

The rows are always placed at the end of the list.


|ASC|DESC|
|-|-|
|![Screenshot 2024-08-26 at 12 39 28](https://github.com/user-attachments/assets/a9f7218f-5b93-479e-86c3-601eee48f03d)|![Screenshot 2024-08-26 at 12 39 21](https://github.com/user-attachments/assets/fadbc937-e9be-4c9c-8ddc-002cd06a15ef)|


Note: That `N/A` was there before my changes, is handled by:

````
const formatMetric = (type: InfraAssetMetricType, value: number | undefined | null) => {
  return value || value === 0 ? createInventoryMetricFormatter({ type })(value) : 'N/A';
};
`````

Reply:
Nevermind, the sorting is handled from our side, I can make some workaround there.
I thought it was internally managed by the table component.

Going to fix it and let you know, thank you!